### ISSUE TITLE
[BUG] Default accelerate version is incompatible with DeepSpeed gradient accumulation

### ISSUE THREAD

**Issue by 1844065745** [starVLA/starVLA#116]
#### Description

The `starvla` repository currently pins an outdated version of `accelerate` (`==1.5.2`).  
This version is **incompatible with the DeepSpeed gradient accumulation semantics currently expected by the training code**, leading to incorrect gradient accumulation behavior.

As a result, **gradient accumulation boundaries are not handled correctly when DeepSpeed is enabled**, which directly affects the effective global batch size during training.

---

#### Current Behavior

I am training with the following configuration:
- `Number of GPUs: 2`
- `per_device_batch_size = 4`
- `gradient_accumulation_steps = 32`

The expected global batch size is:
2 (GPUs) × 4 (per-device batch) × 32 (gradient accumulation) = 256

However, when using the default dependency `accelerate==1.5.2`, the observed global batch size is only 8, indicating that gradient accumulation is not functioning as intended under DeepSpeed:
<img width="968" height="36" alt="Image" src="https://github.com/user-attachments/assets/db3f4cce-18e2-4c51-9e06-a002bf48a0c6" />

After upgrading `accelerate` to `1.12.0`, and keeping all other settings unchanged, the effective global batch size becomes 256 as expected:
<img width="1029" height="42" alt="Image" src="https://github.com/user-attachments/assets/b642ec46-e682-4cec-8d2f-7edf717e6211" />

---

#### Reference

This issue has already been fixed upstream in HuggingFace Accelerate PR [#3630](https://github.com/huggingface/accelerate/pull/3630), which unifies and correctly handles gradient accumulation boundaries for both DeepSpeed and non-DeepSpeed backends.

The fix is not present in `accelerate==1.5.2`, but is included in newer releases (post [**v1.8.0**](https://github.com/huggingface/accelerate/releases/tag/v1.8.0)).
